googler

```diff
@@ -2291,6 +2298,10 @@ class Result(object):
                 print(' ' * (indent + pre) + metadata)
 
         if colors:
+            # Start from the last match, as inserting the bold characters changes the offsets.
+            matches.reverse()
+            for match in matches:
```

Comment:
`for match in reversed(matches)` is enough, and doesn't mutate the argument, which is an unintended side effect.

Reply:
Actually, should use `reversed(matches or [])` now that `matches` could be `None`.